chia/types/blockchain_format/coin.py

```diff
@@ -33,13 +41,16 @@ def name_str(self) -> str:
 
     @classmethod
     def from_bytes(cls, blob):
-        parent_coin_info = blob[:32]
-        puzzle_hash = blob[32:64]
-        amount = int_from_bytes(blob[64:])
-        return Coin(parent_coin_info, puzzle_hash, uint64(amount))
-
-    def __bytes__(self):
-        return self.parent_coin_info + self.puzzle_hash + int_to_bytes(self.amount)
+        # this function is never called. We rely on the standard streamable
```

Comment:
I'm not sure I agree with asserting False here. `from_bytes` is supposed to be the same as `parse`, and similarly, `__bytes__` is supposed to be the same as `stream`.

Why not just use the serialization here?

Reply:
Because the previous version had a different behaviour for __bytes__, specifically to make get_hash() compute the hash over the clvm encoding instead of the streamable format. Asserting false makes sure there aren't any other lingering calls to this that depends on the previous behaviour.